app/cypress/integration/upload.spec.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
+/// <reference types="Cypress" />
+
+context('Homepage', () => {
+  beforeEach(function () {
+    const mockedDateString = '2022-10-10';
+    const mockedDate = new Date(mockedDateString);
+    cy.useMockedTime(mockedDate);
+    cy.sqlFixture('dev/001_intake');
+    cy.visit('/');
+  });
+
+  // Commenting out radio inputs until we pass in proper names or ids to select from
+
+  it('should start and fill the first page of the form', () => {
+    cy.get('body').happoScreenshot({ component: 'Applicant Landing Page' });
+
+    cy.get('h1').contains('Welcome');
+
+    cy.get('a').contains('program details');
+
+    // Todo: find a way around using these wait
+    cy.wait(2000);
+
+    cy.get('button').contains('Go to dashboard').click();
+
+    cy.url().should('contain', '/dashboard');
+
+    // Dashboard page
+    cy.get('h1').contains('Dashboard');
+
+    cy.get('body').happoScreenshot({ component: 'Dashboard Page' });
+
+    cy.get('a').contains('Edit').click();
+    cy.wait(2000);
+    cy.get('a').contains('Supporting documents').click();
+    cy.wait(2000);
+    
+    // cy.visit('http://localhost:3000/applicantportal/form/1/12');
+    cy.get('#root_copiesOfRegistration-btn').click(); 
+    cy.get('[data-testid=file-test]').first().selectFile('cypress/fixtures/doc.txt', { force: true });
```

Comment:
Can we use https://testing-library.com/docs/cypress-testing-library/intro/ to select elements in cypress tests? Using ids doesn't match the [testing-library guiding principles](https://testing-library.com/docs/guiding-principles), which we're trying to follow

Reply:
hmm .. our main Cypress test is full of ids: https://github.com/bcgov/CONN-CCBC-portal/blob/main/app/cypress/integration/index.spec.js#L59
Besides, all out upload controls on the page look exactly same: 
```<input data-testid="file-test" type="file" required="" style="display: none;">```

Reply:
I will change `cy.get('#root_copiesOfRegistration-btn').click(); ` to `cy.get('input[id="root_copiesOfRegistration-btn"]).click(); ` though..

Reply:
Done